contrib/mesos/pkg/queue/delay.go

```diff
@@ -164,6 +188,7 @@ func (q *DelayQueue) pop(next func() *qitem, cancel <-chan struct{}) interface{}
 			select {
 			case <-cancel:
 				item.readd(item)
+				sitAndSpin(&q.cond, ch)
```

Comment:
it's simpler and (in real life usage) probably all that is needed. because we're not creating these queues and throwing them away in the real world, we use them indefinitely. however it's less correct so I think I prefer this version.